tests/unit/test_blob.py

```diff
@@ -456,6 +458,16 @@ def _generate_signed_url_helper(
         else:
             expected_headers = headers
 
+        if virtual_hosted_style_endpoint and version == "v4":
+            expected_headers = headers or {}
+            storage_uri = api_access_endpoint.split("://")
+            api_access_endpoint = (
+                storage_uri[0] + "://" + bucket.name + "." + storage_uri[1]
+            )
```

Comment:
`api_access_endpoint = ("{}://{}.{}".format(storage_uri[0], bucket.name, storage_uri[1])`
makes it also more consistent with the next line.